src/core/algorithms/md/md_verifier/md_verifier.cpp

```diff
@@ -0,0 +1,130 @@
+#include "algorithms/md/md_verifier/md_verifier.h"
+
+#include <algorithm>
+#include <numeric>
+#include <ranges>
+
+#include "algorithms/md/lhs_column_similarity_classifier.h"
+#include "config/equal_nulls/option.h"
+#include "config/exceptions.h"
+#include "config/indices/option.h"
+#include "config/indices/type.h"
+#include "config/names_and_descriptions.h"
+#include "config/option_using.h"
+#include "config/tabular_data/input_table/option.h"
+
+namespace algos::md {
+MDVerifier::MDVerifier() : MdAlgorithm({}) {
+    using namespace config::names;
+    RegisterOptions();
+    MakeOptionsAvailable({kLeftTable, kRightTable});
+}
+
+void MDVerifier::ResetStateMd() {
+    md_holds_ = true;
+}
+
+void MDVerifier::LoadDataInternal() {
+    left_schema_ = std::make_shared<RelationalSchema>(left_table_->GetRelationName());
+    std::size_t const left_table_cols = left_table_->GetNumberOfColumns();
+    for (model::Index i : std::views::iota(model::Index(0), left_table_cols)) {
+        left_schema_->AppendColumn(left_table_->GetColumnName(i));
+    }
+
+    if (right_table_ == nullptr) {
+        right_schema_ = left_schema_;
+
+    } else {
+        right_schema_ = std::make_unique<RelationalSchema>(right_table_->GetRelationName());
+        std::size_t const right_table_cols = right_table_->GetNumberOfColumns();
+        for (model::Index i : std::views::iota(model::Index(0), right_table_cols)) {
+            right_schema_->AppendColumn(right_table_->GetColumnName(i));
+        }
+    }
+}
```

Comment:
1. `right_schema_` is defined as `std::shared_ptr` so why do we fill initialize it with `std::make_unique`?
2. If we replace `make_unique` with `make_shared`, now all of a sudden those are the similar functions, so they can be simplified as:
```suggestion
auto CreateSchema(config::InputTable const& table) {
    auto schema = std::make_shared<RelationalSchema>(table->GetRelationName());
    std::size_t const cols = table->GetNumberOfColumns();
    for (model::Index i : std::views::iota(model::Index(0), cols)) {
        schema->AppendColumn(table->GetColumnName(i));
    }
    return schema;
}

void MDVerifier::LoadDataInternal() {
    left_schema_ = CreateSchema(left_table_);
    right_schema_ = right_table_ ? CreateSchema(right_table_) : left_schema_;
}
```